topics/statistics/tutorials/RNN/tutorial.md

```diff
@@ -0,0 +1,374 @@
+vector whose third element is 1 and other elements are 0. As you can see only one element in the vector is 1 and the rest are 0's. The same 
+concept applies if the size of the vocabulary is N.    
+
+![Alternative text](../../images/OHE.gif "One hot encoding (OHE) representation")
+
+The problem with OHE is that for very large vocabulary sizes (say, 100,000 words) it requires tremendous amount of storage. Also, it has no 
+concept of word similarity.   
+
+## Word2Vec
+
+In Word2Vec, each word is represented as an n dimensional vector (n being much smaller than vocabulary size), such that the words that have 
+similar meanings are closer to each other in the vector space, and words that don't have a similar meaning are farther apart. Words are 
+considered to have a similar meaning if they co-occur often in documents. There are 2 Word2Vec architectures, one that predicts the probability 
+of a word given the surrounding words (Continuous BOW), and one that given a word predicts the probability of the surrounding words (Continuous 
+skip-gram).
+
+In this tutorial, we find an n dimensional representation of the IMDB movie review words, not based on word meanings, but based on how they
+improve the sentiment classification task. The n dimensional representation is learned by the learning algorithm, simply by reducing the 
+cost function via backpropagation. 
+
+# Get data
+
+> ### {% icon hands_on %} Hands-on: Data upload
+>
+> 1. Create a new history for this tutorial
+> 2. Import the files from [Zenodo](https://zenodo.org/record/4477881#.YBrEQ3dKgvo) or from the shared data library
+>
+>    ```
+>    https://zenodo.org/record/4477881/files/X_test.tsv
+>    https://zenodo.org/record/4477881/files/X_train.tsv
+>    https://zenodo.org/record/4477881/files/y_test.tsv
+>    https://zenodo.org/record/4477881/files/y_train.tsv
+>    ```
+>    {% include snippets/import_via_link.md %}
+>    {% include snippets/import_from_data_library.md %}
+>
+> 3. Rename the datasets as `X_test`, `X_train`, `y_test`, and `y_train` respectively
+>
+>    {% include snippets/rename_dataset.md %}
+>
+> 4. Check that the datatype of all the four datasets is `tabular`. If not, change the dataset's datatype to tabular. 
+>
+>    {% include snippets/change_datatype.md datatype="datatypes" %}
+>
+{: .hands_on}
+
+# Sentiment Classification of IMDB movie reviews with RNN
+
+In the section, we define a RNN and train it using IMDB movie reviews training dataset. The goal is to learn a model such that given the 
+words in a review we can predict whether the review was positive or negative. We then evaluate the trained RNN on the test dataset 
+and plot the confusion matrix.
+
+### **Create a deep learning model architecture**
+
+> ### {% icon hands_on %} Hands-on: Model config
+>
+> 1. **Create a deep learning model architecture** using Keras {% icon tool %} with the following parameters:
```

Comment:
If you could use the new tool linking syntax it would be really nice for users!

We have a [guide on how to do it](https://training.galaxyproject.org/training-material/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html#tool-links), but in short just switch to `{% tool [Create a deep learning model architecture](toolshed.g2.bx.psu.edu/repos/..../your-tool/1.0.0+galaxy0) %}`

It can also be done for the upload steps above.

Reply:
@hexylena I added the new tool linking syntax to line 248. But after running 'make serve' and viewing the page in a browser, I just see a link that does not work. Could you please take a look at the code on line 248 and see if you see any issues? BTW, I tried grep'ing the repo to find another example, but could not find one.

Reply:
FYI, I updated all instances to use new toll linking syntax.

Reply:
Here's another example of it being used: https://github.com/galaxyproject/training-material/blob/master/topics/visualisation/tutorials/circos/tutorial.md, there are a couple, we're slowly updating things :)

The link is *expected* to not work when you're just viewing the training (hovering over it should give the tool ID). It only is active when viewed from within Galaxy.

Your changes look exactly correct, I can check locally later.